src/countBy.ts

```diff
@@ -0,0 +1,57 @@
+import { purry } from "./purry";
+
+/**
+ * Categorize and count elements in an array using a defined callback function. The transformation function applies to each element in the array to determine its category and then counts how many elements fall into each category.
```

Comment:
Please add newlines so that all lines are up to 80 characters long. This makes viewing the code side-by-side easier (for example, when reviewing it 😉 ).

Reply:
i feel like there's a prettier rule that can enforce this, if we wanted to

Reply:
Sorry about that, VS Code displays the content in multiple lines so I missed that.